torchaudio/functional.py

```diff
@@ -1287,15 +1287,26 @@ def phaser(
     delay_pos = 0
     mod_pos = 0
 
-    output_waveform = torch.zeros_like(waveform, dtype=dtype, device=device)
+    # TODO: Autograd support?
+    output_waveform_pre_gain_list = []
+    waveform = waveform * gain_in
```

Comment:
Applying gain_in here instead of repeatedly at the beginning of the loop also makes a big difference in performance.